core/db_models/EEM_Event_Question_Group.model.php

```diff
@@ -42,11 +52,26 @@ protected function __construct($timezone = null)
     /**
      * Decides whether to use the 'EQG_primary' or newer 'EQG_additional' for use in queries, based on whether
      * this is concerning primary attendees or additional attendees.
+     * If 1, true, or "primary" is passed in, returns EQG_primary. If 0, false, or "additional" is passed in, returns
+     * EQG_additional.
      * @since $VID:$
-     * @param $for_primary_attendees
+     * @param string|boolean|int $context
      * @return string
      */
-    public function field_name_for_category($for_primary_attendees) {
-        return $for_primary_attendees ? 'EQG_primary' : 'EQG_additional';
+    public function fieldNameForContext($context)
+    {
+        switch (true) {
+            case 'additional' === $context:
+            case false === $context:
+            case 0 === $context:
```

Comment:
oh ya I'll add a comment about that. It's because switch statements don't do strict comparisons, so it was evaluating `true == 'additional'` to be true and not working. I'm open to alternative ways of doing this (also I noticed these are yoda conditionals, oups!)